### DISCUSSION TITLE
Endpoint should be fully qualified endpoint (Parameter 'Endpoint')

### DISCUSSION THREAD

**Discussion by RicoSaupeB** [piotr-rojek/devopsifyme-sbemulator#13] [General]
Hi. I have a console app to just connect to my local azure service bus emulator running in docker. I exported and installed the certificate as suggested but if I run the code I get some exceptions.

1. with this endpoint "Endpoint=sb://localhost/;SharedAccessKey...."

![image](https://user-images.githubusercontent.com/83805528/233023662-7a0443b1-71c7-4e42-82f6-79430b8532df.png)

2. using this endpoint ""Endpoint=sb://127.0.0.1/;SharedAccessKeyName=all...."

![image](https://user-images.githubusercontent.com/83805528/233023848-5b45f4a6-bd2a-491a-843b-6899a163b39e.png)

Innerexception: 

System.Security.Authentication.AuthenticationException: The remote certificate is invalid according to the validation procedure: RemoteCertificateNameMismatch
   at Microsoft.Azure.ServiceBus.Core.MessageSender.OnSendAsync(IList`1 messageList)
   at Microsoft.Azure.ServiceBus.RetryPolicy.RunOperation(Func`1 operation, TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.RetryPolicy.RunOperation(Func`1 operation, TimeSpan operationTimeout)
   at Microsoft.Azure.ServiceBus.Core.MessageSender.SendAsync(IList`1 messageList)

Is it because I am using 127.0.0.1 as replacement for localhost? But with localhost I get the above error. What could be the reason here? 
Thanks for any input. 

the code I am using is pretty simple. 

![image](https://user-images.githubusercontent.com/83805528/233025006-7b72db03-104f-4a89-8c1e-85f4b7b23207.png)

**Comment by piotr-rojek** [comment]
Hey! Look [here](https://github.com/piotr-rojek/devopsifyme-sbemulator/blob/main/docs/install-docker.md) for the list of supported hosts. I think I have seen this behavior with older SDK versions; try upgrading to the latest nuget version. Alternatively, you can use devopsifyme-local.servicebus.windows.net after adding it to the /etc/hosts pointing at 127.0.0.1. 
/Piotr

**Comment by RicoSaupeB** [comment]
Thanks for the hint. I updated the nuget package to the latest version. 

`
<ItemGroup>
        <PackageReference Include="Microsoft.Azure.ServiceBus" Version="5.2.0" />
</ItemGroup>`

I changed host file to that
127.0.0.1	devopsifyme-local.servicebus.windows.net

this led to the certificate problem again. 
using this endpoint: "Endpoint=sb://devopsifyme-local.servicebus.windows.net/;SharedAccess..."

![image](https://user-images.githubusercontent.com/83805528/233281887-de185d9d-0fad-4d0a-b2cc-61c5e3675bb2.png)


using that entry in the host file 
localhost	devopsifyme-local.servicebus.windows.net

using the same endpoint as above.

This leads to a unknown host exception

![image](https://user-images.githubusercontent.com/83805528/233282982-7dab1c08-640e-4c3a-ac2f-f84406c4640d.png)

anything that I missed?

**Reply by piotr-rojek** [reply]
Since you are on the latest nuget version, how does it behave with just "localhost" after the version bump?

**Reply by RicoSaupeB** [reply]
It still complains about the full qualified endpoint. 

running everything in docker and portal is working as well. Sending and receiving messages is fine.

![image](https://user-images.githubusercontent.com/83805528/233578452-2d54c9a7-2883-4422-8d6c-0270ce07266e.png)

**Comment by piotr-rojek** [comment]
Is it an option for you to upgrade to the latest Azure.Messaging.ServiceBus nuget, as it does not have this limitation requiring FQDN. Microsoft.Azure.ServiceBus that you are using at the moment is deprecated as of 11/2020- see https://www.nuget.org/packages/Microsoft.Azure.ServiceBus. 

If you can't upgrade, then let me know, maybe I need to tweak sth in the certificate generation script here: https://github.com/piotr-rojek/devopsifyme-sbemulator/blob/main/src/ServiceBusEmulator.Host/Dockerfile 

You can also generate your own server certificate that you would then trust, as described here https://github.com/piotr-rojek/devopsifyme-sbemulator/blob/main/docs/configuration.md#bring-your-own-certificate

**Comment by RicoSaupeB** [comment]
Thanks for the hint. I indeed took an older example code. Using the newer library it was working fine. Thanks for your support here.
